localeredirector/LocaleRedirectorPlugin.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace Craft;
+
+class LocaleRedirectorPlugin extends BasePlugin
+{
+	public function getName()
+	{
+		return Craft::t('Locale Redirector');
+	}
+
+	public function getDescription()
+	{
+		return Craft::t('Detects the language preferences set by the user, via cookie or browser settings, and redirects to the appropriate locale, if it exists.');
+	}
+
+	public function getVersion()
+	{
+		return '1.0.0';
+	}
+
+	public function getDeveloper()
+	{
+		return 'Jeremy Frank';
+	}
+
+	public function getDeveloperUrl()
+	{
+		return 'http://viget.com';
+	}
+
+	public function getDocumentationUrl()
+	{
+		return 'https://github.com/vigetlabs/craft-localeredirector';
+	}
+
+	public function getReleaseFeedUrl()
+	{
+		return 'https://raw.githubusercontent.com/vigetlabs/craft-localeredirector/master/releases.json';
+	}
+
+	public function init()
+	{
+		// redirects only take place out of the CP
+		if(!craft()->isConsole() && craft()->request->isSiteRequest() && !craft()->request->isLivePreview()) {
+			$currentLocale = craft()->i18n->getLocaleById(craft()->language);
+			$localeCookie = isset($_COOKIE['locale']) ? $_COOKIE['locale'] : null;
+			$browserLanguageMatch = craft()->getTranslatedBrowserLanguage();
```

Comment:
Originally, I was storing `craft()->request->getBrowserLanguages()` to a variable, and then manually finding a match between that and `craft()->i18n->getSiteLocaleIds()`, but alas, Craft already has this built right into **[getTranslatedBrowserLanguage()](https://craftcms.com/classreference/etc/web/WebApp#getTranslatedBrowserLanguage-detail)**!